client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -272,7 +302,18 @@ private final class PoolManager[F[_], A <: Connection[F]](
     * @return An effect of Unit
     */
   override def invalidate(connection: A): F[Unit] =
-    F.delay(disposeConnection(connection.requestKey, Some(connection)))
+    F.delay(synchronized {
+      decrConnection(connection.requestKey)
+      if (!connection.isClosed) connection.shutdown()
+      findFirstAllowedWaiter match {
```

Comment:
This could be consolidated with some code in releaseNonRecyclable.  Just trying to get this working right now, to refactor more later.